pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -480,21 +480,9 @@ func registerHandlers(informer cache.SharedIndexInformer, q queue.Instance, otyp
 		})
 }
 
-// compareEndpoints returns true if the two endpoints are the same in aspects Pilot cares about
-// This currently means only looking at "Ready" endpoints
-func compareEndpoints(a, b *v1.Endpoints) bool {
-	if len(a.Subsets) != len(b.Subsets) {
-		return false
-	}
-	for i := range a.Subsets {
-		if !portsEqual(a.Subsets[i].Ports, b.Subsets[i].Ports) {
-			return false
-		}
-		if !addressesEqual(a.Subsets[i].Addresses, b.Subsets[i].Addresses) {
-			return false
-		}
-	}
-	return true
+// deepEqual is an equality func that uses costly reflect.DeepEqual to check equality.
+func deepEqual(a, b interface{}) bool {
```

Comment:
nit: why not pass `reflect.DeepEqual` directly?

Reply:
makes sense. Changed it